durable-storage/src/merkle_layer/tree.rs

```diff
@@ -15,87 +15,133 @@ use super::node::delete;
 use super::node::get;
 use super::node::get_mut;
 use super::node::set;
+use super::node_resolver::MavlNodeResolver;
+use super::node_wrapper::MavlNodeWrapper;
 
 /// A key-value store tree with left and right nodes that supports traversal and value retrieval.
 #[derive(Clone, Default, Debug)]
-pub(super) struct Avl {
-    root: Option<Arc<MavlNode>>,
+pub(super) struct Avl<Wrapper: MavlNodeWrapper> {
```

Comment:
```suggestion
pub(super) struct Avl<Wrapper> {
```

This trait bound does nothing.

Reply:
Why do you think so?